src/mngt/ocf_mngt_cache.c

```diff
@@ -1790,6 +1737,8 @@ static void _ocf_mngt_attach_handle_error(
 
 	if (context->flags.volume_inited)
 		ocf_volume_deinit(&cache->device->volume);
+	else
+		ocf_volume_deinit(context->cfg.device.volume);
```

Comment:
shouldn't we rather move the volume back from cache->device->volume to the device config? applies to _ocf_mngt_activate_handle_error as well

Reply:
also it would be nice to have a test for this error handling path. injecting volume in volume_ops::open() would be an easy way to do this in pyocf

Reply:
Realistically we do not expect user to retry attach after first attempt fails, so it would basically force user to destroy volume in error handling path manually.